.azure-pipelines.yml

```diff
@@ -0,0 +1,89 @@
+#
+# Azure Pipelines configuration for building and testing react on Linux, Windows, and macOS.
+#
+
+trigger:
+- master
+
+jobs:
+- job: LinuxBasic
+  displayName: 'Linux: Basic Tests'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/basic-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-basic'
+
+- job: LinuxProd
+  displayName: 'Linux: Prod Tests'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/prod-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-prod'
+
+- job: LinuxFire
+  displayName: 'Linux: React Fire Tests'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/fire-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-fire'    
+
+- job: LinuxArtifacts
+  displayName: 'Linux: Build and Upload Artifacts'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/build-and-upload-artifacts.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-artifacts'        
+
+- job: LinuxCoverage
+  displayName: 'Linux: Test Coverage'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/test-coverage.yml
+
+- job: WindowsTests
+  displayName: 'Windows: Basic Tests'
+  pool:
+    vmImage: 'vs2017-win2016'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - script: |
```

Comment:
I would  move the git config steps before the steps in `setup.yml`. Seems like there's a chance they could add a post-install step that could cause problems if the line endings aren't correct.

Reply:
Moved.